src/components/Excerpt.tsx

```diff
@@ -85,45 +102,78 @@ export type ExcerptProps = {
   overflowThreshold?: number;
 
   /**
-   * Called when the content height exceeds or falls below
+   * Whether a shadow is drawn at the bottom of collapsed content, to hint that
+   * content is being hidden.
+   *
+   * The shadow area can be clicked to expand the container so the content is
+   * fully visible.
+   *
+   * Defaults to `false` for excerpts with inline control, and `true` for
+   * excerpts with external control.
+   */
+  shadow?: boolean;
+
+  /**
+   * If the content should be truncated if its height exceeds
    * `collapsedHeight + overflowThreshold`.
+   *
+   * Use this prop in combination with `onToggleCollapsed` to make this excerpt
+   * a controlled component.
+   *
+   * Defaults to `true`.
    */
-  onCollapsibleChanged?: (isCollapsible: boolean) => void;
+  collapsed?: boolean;
 
   /**
-   * When `inlineControls` is `false`, this function is called when the user
-   * requests to expand the content by clicking a zone at the bottom of the
-   * container.
+   * If this function is provided, it is called when the user requests to expand
+   * the content by clicking the shadowed zone at the bottom of the container
+   * (if `shadow` is `true`) or the inline control (if `inlineControl` is `true`)
    */
   onToggleCollapsed?: (collapsed: boolean) => void;
 
   /**
-   * Additional styles to pass to the inline controls element.
-   * Ignored if inlineControls is `false`.
+   * Called when the content height exceeds or falls below
+   * `collapsedHeight + overflowThreshold`.
    */
-  inlineControlsLinkStyle?: JSX.CSSProperties;
-};
+  onCollapsibleChanged?: (isCollapsible: boolean) => void;
+} & (InlineControlExcerptProps | ExternalControlExcerptProps);
 
 /**
  * A container which truncates its content when they exceed a specified height.
  *
  * The collapsed state of the container can be handled either via internal
- * controls (if `inlineControls` is `true`) or by the caller using the
- * `collapse` prop.
+ * controls (if `inlineControls` is `true`) or by the caller using a custom
+ * control.
  */
 export default function Excerpt({
   children,
-  collapse = false,
+  collapsed,
   collapsedHeight,
-  inlineControls = true,
-  onCollapsibleChanged = noop,
-  onToggleCollapsed = noop,
+  onCollapsibleChanged,
+  onToggleCollapsed,
   overflowThreshold = 0,
-  inlineControlsLinkStyle = {},
+  shadow,
+  ...rest
 }: ExcerptProps) {
-  const [collapsedByInlineControls, setCollapsedByInlineControls] =
-    useState(true);
+  // If `collapsed` is present, treat this as a controlled component
+  const isControlled = typeof collapsed === 'boolean';
+  // Only use this local state if Excerpt is uncontrolled
+  const [uncontrolledCollapsed, setUncontrolledCollapsed] = useState(
+    collapsed ?? true,
```

Comment:
To make this mirror other components which can be either controlled or uncontrolled, you'd need a `defaultCollapsed` prop to set the initial state in the uncontrolled case.